Custom logical types get omitted from the classpath

The `SchemaAssistant` tracks all the jars that the generated code will need on the classpath, including the classes for all logical types. However, it neglects to track the classes that represent logical types derived from `fixed` types. Whenever a logical type is represented by a class not in the Java SDK and not in the same jar as its `Conversion` class, compilation will fail because the class can't be found.

This is an example of a runtime error in the logs.
```
/tmp/generated1750127073899966702/com/linkedin/avro/fastserde/generated/serialization/AVRO_1_11/ExerciseLogicalTypes_GenericSerializer_1478679821.java:33: error: cannot access UID
        convertedValue0 = Conversions.convertToRawType(convertedValue0, this.logicalTypeSchema__692588075, this.logicalTypeSchema__692588075 .getLogicalType(), this.conversion_uid);
                                     ^
  class file for org.example.logging.UID not found
Note: /tmp/generated1750127073899966702/com/linkedin/avro/fastserde/generated/serialization/AVRO_1_11/ExerciseLogicalTypes_GenericSerializer_1478679821.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error
2024-11-19 17:48:56.752 [ERROR] 1885108 --- [ompile-thread-1] m.linkedin.avro.fastserde.FastSerdeCache : Serializer class instantiation exception
```

--- a/fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/SchemaAssistant.java
+++ b/fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/SchemaAssistant.java
@@ -297,9 +297,12 @@ public class SchemaAssistant<T extends GenericData> {
   /* Note that settings abstractType and rawType are not passed to subcalls */
   public JClass classFromSchema(Schema schema, boolean abstractType, boolean rawType, boolean primitiveList,
           boolean allowLogicalTypes) {
-    if (allowLogicalTypes && logicalTypeEnabled(schema)) {
+    if (logicalTypeEnabled(schema)) {
       Class<?> logicalTypeClass = ((Conversion<?>) getConversion(schema.getLogicalType())).getConvertedType();
-      return codeModel.ref(logicalTypeClass);
+      fullyQualifiedClassNameSet.add(logicalTypeClass.getName());
+      if (allowLogicalTypes) {
+        return codeModel.ref(logicalTypeClass);
+      }
     }
 
     JClass outputClass;

